server/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -243,7 +301,21 @@ public boolean hasMappings() {
      * type then the fields will be returned with a type prefix.
      */
     public Set<String> simpleMatchToIndexNames(String pattern) {
-        return mapperService.simpleMatchToFullName(pattern);
+        // TODO remove the duplication with MapperService and FieldTypeLookup
```

Comment:
how are you planning to address this? Should the method in MapperService take the runtime mappings as argument or is that too intrusive of a change?

Reply:
I was hoping to not think about it today. Pushing the runtime mappings over might do. I'll have a look once I get all the tests happy.

Reply:
After looking a little more I wonder if the duplication is that bad - its another check of `isSimpleMatchPattern` when there are runtime fields. Compared to iterating a HashMap it seems ok.